src/opm/material/fluidmatrixinteractions/EclMaterialLawManagerHystParams.cpp

```diff
@@ -0,0 +1,291 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
+// vi: set et ts=4 sw=4 sts=4:
+/*
+  Copyright 2022 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <config.h>
+#include <opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp>
+#include <opm/material/fluidmatrixinteractions/EclEpsGridProperties.hpp>
+
+namespace Opm {
+
+/* constructors*/
+template <class Traits>
+EclMaterialLawManager<Traits>::InitParams::HystParams::
+HystParams(EclMaterialLawManager<Traits>::InitParams& init_params) :
+    init_params_{init_params}, parent_{init_params_.parent_},
+    eclState_{init_params_.eclState_}
+{
+    gasOilParams_ = std::make_shared<GasOilTwoPhaseHystParams>();
```

Comment:
I can be wrong.  During the memory profiling, it looks like these three parameters can be pretty big.  For the SPE11C case, the three Params_  are allocated  59.2 MB each and about 180 MB in total.  And there is no HYSTER related in the SPE11C case. 

Maybe it is good only allocate memory for these when needed. @totto82

Reply:
The hystParams contains the two-phase params and end-point scaling etc so we can not remove it completely. But as I understand the code we currently allocate memory to the hysteresis parameters ++ even though we dont need them. Have you checked the memory profile if HYSTER is activated. Does it change significantly?